pkg/volume/csi/csi_plugin.go

```diff
@@ -159,7 +160,7 @@ func (h *RegistrationHandler) DeRegisterPlugin(pluginName string) {
 
 func (p *csiPlugin) Init(host volume.VolumeHost) error {
 	p.host = host
-
+	PluginHandler.host = host
```

Comment:
Updating this singleton instance could introduce timing issues.  For example, the singleton instance could try to use host before we've initialized it here.

I think it would be better to remove this singleton and use a proper constructor method.  Sorry, this fix is becoming more involved. It might be good to do this refactor in a different PR.

cc @vladimirvivien if you have any better suggestions